### PR TITLE
Dump OS release info, inbox drivers versions to a JSON file

### LINKED ISSUE #19: [Linux] Convert inbox driver version output into json format
After check_inbox_driver completed, an inbox_driver_versions.txt file will be created which includes all inbox driver versions. Now we need to convert the file into JSON format, and which shall also include cloud-init version.

### PR DESCRIPTION
**Pull request by keirazhang**
Fix #19
In check_inbox_driver.yml, it will collect OS release information, kernel version, ovt version, cloud-init version, vmware inbox drivers versions, etc, and then dump the results into a json file.

Tested with RHEL 8.3
```
Test Results (Total: 2, Failed: 0, No Run: 0, Elapsed Time: 00:08:11):
+-----------------------------------------+
| Name               | Status | Exec Time |
+-----------------------------------------+
| check_inbox_driver | Passed | 00:02:04  |
| ovt_verify_status  | Passed | 00:05:42  |
+-----------------------------------------+

rhel-8.3.json:
[
    {
        "Release": "RHEL 8.3",
        "kernel": "4.18.0-240.el8.x86_64",
        "open-vm-tools": "11.1.0.19887 (build-16036546)",
        "vmw_balloon": "1.5.0.0-k",
        "vmw_pvscsi": "1.0.7.0-k",
        "vmw_vmci": "1.1.6.0-k",
        "vmw_vsock_vmci_transport": "1.0.5.0-k",
        "vmwgfx": "2.17.0.0",
        "vmxnet3": "1.5.0.0-k",
        "vsock": "1.0.2.0-k",
        "xorg server": "1.20.8"
    }
]
```

Tested with SLED 15 SP3:
```
Test Results (Total: 2, Failed: 0, No Run: 0, Elapsed Time: 00:07:53):
+-----------------------------------------+
| Name               | Status | Exec Time |
+-----------------------------------------+
| check_inbox_driver | Passed | 00:02:27  |
| ovt_verify_status  | Passed | 00:04:14  |
+-----------------------------------------+

sle-15-sp3.json 
[
    {
        "Release": "SLE 15 SP3",
        "kernel": "5.3.18-51-default",
        "open-vm-tools": "11.2.5.26209 (build-17337674)",
        "vmw_pvscsi": "1.0.7.0-k",
        "vmw_vmci": "1.1.6.0-k",
        "vmw_vsock_vmci_transport": "1.0.5.0-k",
        "vmwgfx": "2.18.0.0",
        "vmxnet3": "1.5.0.0-k",
        "vsock": "1.0.2.0-k"
        "xf86-video-vmware": "13.3.0",
        "xorg server": "1.20.3"
    }
]
```

Tested with Ubuntu 20.10:
```
Test Results (Total: 2, Failed: 0, No Run: 0, Elapsed Time: 00:03:56):
+-----------------------------------------+
| Name               | Status | Exec Time |
+-----------------------------------------+
| check_inbox_driver | Passed | 00:01:46  |
| ovt_verify_status  | Passed | 00:01:44  |
+-----------------------------------------+

ubuntu-20.10.json:
[
    {
        "Release": "Ubuntu 20.10",
        "cloud-init": "20.3-15-g6d332e5c-0ubuntu1",
        "kernel": "5.8.0-50-generic",
        "open-vm-tools": "11.1.5.22735 (build-16724464)",
        "vmw_pvscsi": "1.0.7.0-k",
        "vmw_vmci": "1.1.6.0-k",
        "vmw_vsock_vmci_transport": "1.0.5.0-k",
        "vmwgfx": "2.18.0.0",
        "vmxnet3": "1.5.0.0-k",
        "vsock": "1.0.2.0-k",
        "xorg server": "1.20.9"
    }
]
```

### CODE REVIEW

**Comment by keirazhang** [comment]
Tested with Photon 4.0:
```
Test Results (Total: 3, Failed: 0, No Run: 0, Elapsed Time: 00:06:10):
+-------------------------------------------+
| Name                 | Status | Exec Time |
+-------------------------------------------+
| check_inbox_driver   | Passed | 00:01:25  |
| ovt_verify_status    | Passed | 00:01:20  |
| ovt_verify_uninstall | Passed | 00:02:57  |
+-------------------------------------------+

vmware-photon-os-4.0.json 
[
    {
        "Release": "VMware Photon OS 4.0",
        "cloud-init": "20.4.1",
        "kernel": "5.10.4-13.ph4-esx",
        "open-vm-tools": "11.2.5.26209 (build-17337674)",
        "vmw_balloon": "",
        "vmw_pvscsi": "1.0.7.0-k",
        "vmw_vmci": "1.1.6.0-k",
        "vmw_vsock_vmci_transport": "1.0.5.0-k",
        "vmwgfx": "2.18.0.0",
        "vmxnet3": "1.5.0.0-k",
        "vsock": "1.0.2.0-k"
    }
]
```

**Comment by keirazhang** [comment]
Tested with Flatcar:
```
Test Results (Total: 2, Failed: 0, No Run: 0, Elapsed Time: 00:02:28):
+-----------------------------------------+
| Name               | Status | Exec Time |
+-----------------------------------------+
| check_inbox_driver | Passed | 00:00:58  |
| ovt_verify_status  | Passed | 00:01:01  |
+-----------------------------------------+

flatcar-2765.2.2.json 
[
    {
        "Release": "Flatcar 2765.2.2",
        "kernel": "5.10.25-flatcar",
        "open-vm-tools": "11.2.5.26209 (build-17337674)",
        "vmw_pvscsi": "1.0.7.0-k",
        "vmw_vmci": "1.1.6.0-k",
        "vmw_vsock_vmci_transport": "1.0.5.0-k",
        "vmwgfx": "2.15.0.0",
        "vmxnet3": "1.4.17.0-k",
        "vsock": "1.0.2.0-k"
    }
]
```

Tested with Debian 10.7
```
Test Results (Total: 2, Failed: 0, No Run: 0, Elapsed Time: 00:04:42):
+-----------------------------------------+
| Name               | Status | Exec Time |
+-----------------------------------------+
| check_inbox_driver | Passed | 00:01:14  |
| ovt_verify_status  | Passed | 00:03:03  |
+-----------------------------------------+

debian-10.7.json 
[
    {
        "Release": "Debian 10.7",
        "kernel": "4.19.0-13-686-pae",
        "open-vm-tools": "10.3.10.10540 (build-12406962)",
        "vmw_pvscsi": "1.0.7.0-k",
        "vmw_vmci": "1.1.6.0-k",
        "vmw_vsock_vmci_transport": "1.0.5.0-k",
        "vmwgfx": "2.15.0.0",
        "vmxnet3": "1.4.17.0-k",
        "vsock": "1.0.2.0-k",
        "xf86-video-vmware": "13.3.0-2",
        "xorg server": "1.20.4"
    }
]
```

File: linux/check_inbox_driver/check_inbox_driver.yml [hunk 1]
```diff
@@ -13,34 +13,59 @@
...
               set_fact:
-                versions: "{{ versions | combine({'open-vm-tools': ovt_version.stdout}) }}"
+                inbox_drivers_versions: "{{ inbox_drivers_versions | combine({'open-vm-tools': ovt_version.stdout}) }}"
           when: vmtools_is_ovt is defined and vmtools_is_ovt
 
-        - name: Get drivers version in VMware Photon OS
+        # Get cloud-init version
+        - include_tasks: ../utils/cloudinit_pkg_check.yml
+
+        - name: "Collect cloud-inint version"
```
Comment (Tomorrow9) [inline linux/check_inbox_driver/check_inbox_driver.yml:61]: A typo of "cloud-init".

File: linux/gosv_testcase_list.yml [hunk 2]
```diff
@@ -2,13 +2,13 @@
 # SPDX-License-Identifier: BSD-2-Clause
 ---
 - import_playbook: deploy_vm/deploy_vm.yml
+- import_playbook: check_inbox_driver/check_inbox_driver.yml
```
Comment (Tomorrow9) [inline linux/gosv_testcase_list.yml:5]: Do we need to check/get driver info after ovt installed or updated?

Comment (keirazhang) [reply linux/gosv_testcase_list.yml:5]: New change:
If no open-vm-tools is installed while running this case, and ovt_verify_install installs an ovt from OS image or OS official repo, that ovt version will be updated into os release info json file by log plubin.

**Comment by keirazhang** [comment]
CentOS 7.9 Minimal
```
VM information:
+-----------------------------------------------------+
| VM Name             | qiz-214-centos-7-0430         |
+-----------------------------------------------------+
| VM IP               | 10.117.16.147                 |
+-----------------------------------------------------+
| Guest OS Type       | CentOS 7.9 x86_64             |
+-----------------------------------------------------+
| VM Tools            | 11.0.5.17716 (build-15389592) |
+-----------------------------------------------------+
| Cloud-Init          | 19.4                          |
+-----------------------------------------------------+
| Guest ID            | centos7_64Guest               |
+-----------------------------------------------------+
| Hardware Version    | vmx-19                        |
+-----------------------------------------------------+

Test Results (Total: 4, Failed: 0, No Run: 0, Elapsed Time: 00:24:41):
+--------------------------------------------------------+
| Name                              | Status | Exec Time |
+--------------------------------------------------------+
| deploy_vm_efi_paravirtual_vmxnet3 | Passed | 00:12:36  |
| check_inbox_driver                | Passed | 00:01:27  |
| ovt_verify_install                | Passed | 00:04:36  |
| gosc_cloudinit_dhcp               | Passed | 00:05:43  |
+--------------------------------------------------------+

centos-7.9.json 
[
    {
        "Release": "CentOS 7.9",
        "open-vm-tools": "11.0.5.17716 (build-15389592)",
        "cloud-init": "19.4",
        "kernel": "3.10.0-1160.el7.x86_64",
        "vmw_pvscsi": "1.0.7.0-k",
        "vmw_vmci": "1.1.6.0-k",
        "vmw_vsock_vmci_transport": "1.0.5.0-k",
        "vmwgfx": "2.15.0.0",
        "vmxnet3": "1.4.17.0-k",
        "vsock": "1.0.2.0-k"
    }
]
```

File: linux/open_vm_tools/install_ovt.yml [hunk 3]
```diff
@@ -52,3 +52,18 @@
 # Eject CDROM /dev/sr0 from guest
 - include_tasks: ../utils/eject_cdrom_in_guest.yml
```
Comment (Tomorrow9) [inline linux/open_vm_tools/install_ovt.yml:53]: Is there a when condition missed here, "linux_ovt_repo_url is undefined or not linux_ovt_repo_url"?

File: linux/open_vm_tools/install_ovt.yml [hunk 4]
```diff
@@ -52,3 +52,18 @@
...
+# Reboot and Wait for tools running
+- include_tasks: ../utils/vm_reboot.yml
+
+# Get VMware Tools version and build
+- include_tasks: ../utils/get_guest_ovt_version_build.yml
+
+# Get the OS built-in open-vm-tools version
+- name: "Set guest inbox open-vm-tools version"
+  set_fact:
+    guest_inbox_ovt_version: "{{ vmtools_info_from_vmtoolsd }}"
```
Comment (Tomorrow9) [inline linux/open_vm_tools/install_ovt.yml:65]: Here is there a condition that if using online official repo, the VMware tools is updated, not the built-in OVT?

File: linux/open_vm_tools/install_ovt.yml (hunk 3 above)
Comment (keirazhang) [reply linux/open_vm_tools/install_ovt.yml:53]: Fixed.

**Review by Tomorrow9 — APPROVED** [review]

**Pull request merged**